### PR TITLE
Hotfix for savegame compatibility #4273

### LINKED ISSUE #4273: Incompatible b21 savegame
- Open the savegame from https://bugs.launchpad.net/widelands/+bug/1799373/comments/20 with build 21. This will work without problem.
- Resave the savegame with b21.
- Try to load it in current master:
```
widelands: ../src/scripting/lua_map.h:800: const Widelands::TerrainDescription *LuaMaps::LuaTerrainDescription::get() const: Assertion `terraindescr_ != nullptr' failed.
```

### PR DESCRIPTION
**Pull request by matthiakl**
Fixes #4273
The reported error is straight forward, but afterwards I get some invalid wares in the input queues.
Not sure whether these can be ignored or need a proper fix. A quick test didn't show any problems, but I simply got lost in the map ;)

### CODE REVIEW

File: src/logic/playercommand.cc [hunk 1]
```diff
@@ -1302,12 +1302,16 @@ void CmdSetInputMaxFill::execute(Game& game) {
...
+			try {
+				b->inputqueue(index_, type_, nullptr).set_max_fill(max_fill_);
+				if (upcast(Warehouse, wh, b)) {
+					if (PortDock* p = wh->get_portdock()) {
+						// Update in case the expedition was ready previously and now lacks a ware again
+						p->expedition_bootstrap()->check_is_ready(game);
+					}
 				}
+			} catch (const std::exception& e) {
+				log_err("Skipped CmdSetInputMaxFill command: %s", e.what());
```
Comment (Noordfrees) [inline src/logic/playercommand.cc:1316]: Please add a TODO that this can happen only after loading a b21 save and should no longer be caught after v1.0

**Review by Noordfrees — COMMENTED** [review]
Tested and working, very good :)

**Review by Noordfrees — APPROVED** [review]

**Comment by hessenfarmer** [comment]
> 
> 
> Fixes #4273
> The reported error is straight forward, but afterwards I get some invalid wares in the input queues.

What does that mean? We had an issue like this before due to changes in the index of our wares. Might be due to the addition of amazon wares. 
We need to be sure to not introduce a bug here. do you have a screenshot?

**Comment by Noordfrees** [comment]
I had checked that and it's just a saved playercommand that fails to execute, so it's not worth the trouble of fixing this and the command is allowed to fail silently. The game can be played fine.

**Comment by hessenfarmer** [comment]
> 
> 
> I had checked that and it's just a saved playercommand that fails to execute, so it's not worth the trouble of fixing this and the command is allowed to fail silently. The game can be played fine.

ok then. it was just the description that got me alerted

**Comment by matthiakl** [comment]
> We need to be sure to not introduce a bug here. do you have a screenshot?

A screenshot of what?
I did a quick look, whether I can find the problematic wares, but without luck. 
Some of the output:
```
[00:00:28.450 real] ERROR: Skipped CmdSetInputMaxFill command: [/mnt/storage/Programme/widelands/src/logic/map_objects/tribes/productionsite.cc:422] atlanteans_coalmine (43275) has no InputQueue for unknown ware 84
[00:00:28.450 real] ERROR: Skipped CmdSetInputMaxFill command: [/mnt/storage/Programme/widelands/src/logic/map_objects/tribes/productionsite.cc:422] atlanteans_coalmine (43275) has no InputQueue for unknown ware 85
[00:00:28.450 real] ERROR: Skipped CmdSetInputMaxFill command: [/mnt/storage/Programme/widelands/src/logic/map_objects/tribes/productionsite.cc:425] atlanteans_coalmine (43275) has no InputQueue for ware 37: trident_double
[00:00:28.450 real] ERROR: Skipped CmdSetInputMaxFill command: [/mnt/storage/Programme/widelands/src/logic/map_objects/tribes/productionsite.cc:422] atlanteans_goldmine (43345) has no InputQueue for unknown ware 84
[00:00:28.450 real] ERROR: Skipped CmdSetInputMaxFill command: [/mnt/storage/Programme/widelands/src/logic/map_objects/tribes/productionsite.cc:422] atlanteans_goldmine (43345) has no InputQueue for unknown ware 85
[00:00:28.450 real] ERROR: Skipped CmdSetInputMaxFill command: [/mnt/storage/Programme/widelands/src/logic/map_objects/tribes/productionsite.cc:425] atlanteans_goldmine (43345) has no InputQueue for ware 37: trident_double
```
And the whole, if you are interested: 
[out.txt](https://github.com/widelands/widelands/files/6038753/out.txt)

**Comment by Noordfrees** [comment]
>  I did a quick look, whether I can find the problematic wares, but without luck.

The warnings are about the nonexistence of these wares :D What happened here is that shortly before saving the game in b21, the AI players sent a number of playercommands to set the ware capacity for ware X in building Y to Z. After loading with this branch, index X refers to a *different* ware for which the building may not have a use. It would be possible to write a list of all wares with their indices as they appeared in b21 so they can be mapped to the new indices on loading, but this would be a lot of effort for extremely little gain, so the malformed loaded commands are just allowed to be ignored.

**Pull request merged**